src/js/components/Header/Queue.react.js

```diff
@@ -31,9 +34,26 @@ export default class Queue extends Component {
         };
     }
 
-    render() {
+    queueContent(shownQueue) {
+        const self = this;
+        return shownQueue.map((track, index) => {
+            return (
+                <QueueItem
+                    index={ index }
+                    track={ track }
+                    dragged={ index === self.state.draggedTrack }
+                    draggedOver={ index === self.state.draggedOverTrack }
+                    draggedOverAfter={ index === self.state.draggedOverTrack && !self.state.draggedBefore }
+                    onDragStart={ self.dragStart.bind(self, index) }
+                    onDragEnd={ self.dragEnd.bind(self) }
+                    onRemove={ self.removeFromQueue.bind(null, index) }
+                    onPlay={ self.play.bind(null, self.props.queueCursor + index + 1) }
+                />
```

Comment:
Here it's ok to use `eventListeners` because of the component scope/drag management :)